### ISSUE TITLE
Gradient Scaling With Pipeline Parallelism

### ISSUE THREAD

**Issue by windsornguyen** [pytorch/torchtitan#803]
The idiomatic way to perform gradient scaling is something like this:
```python
preds = model(inputs)
loss = loss_fn(preds, targets)
scaler.scale(loss).backward()
```

Given that the current PyTorch PP API handles the backward pass *internally*, I find it difficult to do gradient scaling under a PP regime.

Is there currently a good way to do gradient scaling with Pipeline Parallelism? And if not, will the Pipeline Parallelism API support gradient scaling in the near-term future?

**Comment by mori360** [comment]
> scaler.scale(loss).backward()

In TorchTian, we have gradient scaling with backward ahead, could I have more information on `scaler.scale(loss).backward()` here?
